File: server/routes/images.js
const express = require("express")
const s3 = require("../s3")
const upload = require("../upload")

const router = express.Router()
const BUCKET_NAME = process.env.BUCKET_NAME

// Uploading images on the mobile app
router.post("/", upload.single("image"), async (req, res) => {
  const file = req.file
  if (file) {
    try {
      res.json({ key: file.key, url: file.location })
      console.log(`Uploaded image "${file.key}"`)
    } catch (e) {
      console.log(e)
    }
  } else {
    console.log("Failed uploading image")
  }
})

// Uploading temporary images on website
router.post("/upload", upload.single("image"), async (req, res) => {
  const file = req.file
  if (file) {
    try {
      const presignedUrl = await s3.getSignedUrlPromise("getObject", {
        Bucket: BUCKET_NAME,
        Key: file.key,
      })
      res.json({ url: presignedUrl })
      console.log(`Uploaded temporary image "${file.key}"`)
    } catch (e) {
      console.log(e)
    }
  } else {
    console.log("Failed uploading temporary image")
  }
})

router.delete("/:key", async (req, res) => {
  const { key } = req.params
  await s3.deleteObject({ Bucket: BUCKET_NAME, Key: key }).promise()
  console.log("Deleted image")
})

module.exports = router

File: models/routers/images.py
import os
from io import BytesIO
from fastapi import APIRouter, UploadFile, File
from s3 import s3

router = APIRouter(prefix="/images", tags=["images"])
BUCKET_NAME = os.environ["BUCKET_NAME"]


# Upload image from mobile app to IBM Cloud Object Storage
@router.post("")
async def upload_image(image: UploadFile = File(...)):
    key = os.path.basename(image.filename)
    url = f"{os.environ['ENDPOINT']}/{BUCKET_NAME}/{key}"
    try:
        contents = await image.read()
        with BytesIO(contents) as data:
            s3.upload_fileobj(data, BUCKET_NAME, key)
        print(f"Uploaded image {key}")
    except Exception as e:
        print(e)
        print(f"Failed to upload image {key}")

    return {"key": key, "url": url}


# Uploads temporary images from the website to IBM Cloud Object Storage
@router.post("/upload")
def upload_temporary_image(image: UploadFile = File(...)):
    pass


# Delete image from IBM Cloud Object Storage
@router.delete("/{key}")
def delete_image(key: str):
    try:
        s3.delete_object(Bucket=BUCKET_NAME, Key=key)
        print(f"Deleted image {key}")
    except Exception as e:
        print(e)
        print(f"Failed to delete image {key}")
